src/main/java/com/github/manolo8/darkbot/backpage/NovaManager.java

```diff
@@ -0,0 +1,205 @@
+package com.github.manolo8.darkbot.backpage;
+
+import com.github.manolo8.darkbot.backpage.nova.Agent;
+import com.github.manolo8.darkbot.backpage.nova.NovaData;
+import com.github.manolo8.darkbot.backpage.nova.Perk;
+import com.github.manolo8.darkbot.utils.http.Method;
+import com.google.gson.Gson;
+import com.google.gson.JsonObject;
+
+import java.io.IOException;
+
+public class NovaManager {
+    private final BackpageManager backpageManager;
+    private final NovaData data;
+    private final Gson g;
+    private long lastNovaUpdate;
+
+    public NovaManager(BackpageManager backpageManager) {
+        this.backpageManager = backpageManager;
+        this.data = new NovaData();
+        this.g = new Gson();
+    }
+
+    public boolean update(long expiryTime) {
+        try {
+            if (System.currentTimeMillis() <= lastNovaUpdate + expiryTime) return false;
+
+            data.getRosterList().forEach((k, v) -> v.setForRemoval(true));
+
+            updateActiveCaptain(0);
+            updateResource();
+            updateRosterList();
+
+            data.getRosterList().values().removeIf(Agent::getForRemoval);
+            lastNovaUpdate = System.currentTimeMillis();
+
+            return true;
+        } catch (Exception e) {
+            e.printStackTrace();
+        }
+        return false;
+    }
+
+    public NovaData getData() {
+        return this.data;
+    }
+
+
+    /**
+     * This method is used to unequip, equip and get current captain
+     * @param id
+     * @throws IOException
+     */
+    public boolean updateActiveCaptain(int id) throws IOException {
```

Comment:
id? what id?

Captain id? if so name it properly, and if you're doing javadocs explain what the id is in the `@param` don't just leave it empty